build-tools/packages/build-cli/src/commands/generate/typetests.ts

```diff
@@ -196,31 +196,53 @@ declare type MakeUnusedImportErrorsGoAway<T> = TypeOnly<T> | MinimalType<T> | Fu
 /**
  * Tries to find the path to types for a given API level, falling back to another API level (typically public) if the
  * requested one is not found. The paths returned are relative to the package.
+ *
+ * @remarks
+ * This implementation loosely follows TypeScript's process for finding types as described at
+ * {@link https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-main-and-types}. If an export
+ * map is found, the `types` and `typings` field are ignored. If an export map is not found, then the `types`/`typings`
+ * fields will be used as a fallback _only_ for the public API level (which corresponds to the default export).
+ *
+ * Importantly, this code _does not_ implement falling back to the `main` field when `types` and `typings` are missing,
+ * nor does it look up types from DefinitelyTyped (i.e. \@types/* packages). This fallback logic is not needed for our
+ * packages because we always specify types explicitly in the types field, and types are always included in our packages
+ * (as opposed to a separate \@types package).
  */
 function getTypesPathWithFallback(
 	packageJson: PackageJson,
 	entrypoint: ApiLevel,
 	log: Logger,
 	fallbackEntrypoint?: ApiLevel,
 ): { typesPath: string; entrypointUsed: ApiLevel } {
+	const getTypesPath = (level: ApiLevel): string | undefined => {
```

Comment:
Prefer `function getTypesPath(level: ApiLevel): string | undefined {` to get named function.